src/coreclr/jit/fgflow.cpp

```diff
@@ -291,6 +288,9 @@ flowList* Compiler::fgRemoveRefPred(BasicBlock* block, BasicBlock* blockPred)
         return nullptr;
     }
 
+    assert(block->countOfInEdges() > 0);
```

Comment:
The header comment specifically states:
```
//    -- block->bbRefs is adjusted even if preds haven't been computed. If preds haven't been computed,
//       the preds themselves aren't touched.
```

This matches the behavior of fgAddRefPreds, but here is no longer true.

Are there any diffs due to this?

If we really shouldn't be updating bbRefs beforehand checking fgComputePredsDone, then it seems like we need to have the same behavior in fgAddRefPred.

But I'd like to understand why that behavior exists in the first place before changing it.

Reply:
I'm trying to investigate it now, but the only method I am able to catch it is quite huge 😢

Reply:
I believe there are a number of early phases in the jit that don't keep accurate ref counts. And I also don't think there is anything that depends on them being accurate (for correctness, anyways). You may find you can fix the place that's leading to your current assert, but I suspect there will be more.

So I'd suggest looking at the logic for `fgSplitBlockAfterStmt` or what ever is up-stack when the assert fires, and perhaps adding a bypass if pred lists haven't been maintained.

Over time we'll move the building of pred lists earlier and this will all get sorted out.